execution/engine/eks_engine.go

```diff
@@ -258,19 +259,31 @@ func (ee *EKSExecutionEngine) FetchUpdateStatus(run state.Run) (state.Run, error
 		return run, err
 	}
 
-	var podList *v1.PodList
-	var pod *v1.Pod
-	// Only fetch associated Pod if the the run doesn't have a podName or if the job exited.
-	if job.Status.Failed == 1 || run.PodName == nil {
-		podList, err = ee.getPodList(run)
-		if err == nil && podList != nil && podList.Items != nil && len(podList.Items) > 0 {
-			pod = &podList.Items[len(podList.Items)-1]
+	var mostRecentPod *v1.Pod
+	var mostRecentPodCreationTimestamp metav1.Time
 
-			if run.PodName == nil && pod != nil {
-				run.PodName = &pod.Name
+	podList, err := ee.getPodList(run)
+
+	if err == nil && podList != nil && podList.Items != nil && len(podList.Items) > 0 {
+		ee.log.Log("message", "iterating over pods")
+		// Iterate over associated pods to find the most recent.
+		for _, p := range podList.Items {
+			if mostRecentPodCreationTimestamp.Before(&p.CreationTimestamp) {
+				mostRecentPod = &p
+				mostRecentPodCreationTimestamp = p.CreationTimestamp
 			}
 		}
+
+		// If the run doesn't have an associated pod name yet OR
+		// there is a newer pod (i.e. the old pod was killed),
+		// update it.
+		if mostRecentPod != nil && (run.PodName == nil || strings.Compare(mostRecentPod.Name, *run.PodName) != 0) {
```

Comment:
strings.Compare is it different than doing `mostRecentPod.Name == *run.PodName`?

Reply:
https://www.geeksforgeeks.org/different-ways-to-compare-strings-in-golang/

Reply:
nit